lib/excon/middlewares/redirect_follower.rb

```diff
@@ -2,6 +2,18 @@
 module Excon
   module Middleware
     class RedirectFollower < Excon::Middleware::Base
+      def self.valid_parameter_keys
+        [
+          :redirects_remaining,
+          :redirect_limit
+        ]
+      end
+
+      def request_call(datum)
+        datum[:redirects_remaining] ||= datum[:redirect_limit] ||
```

Comment:
I like how this keeps this out of the default datum (unlike the idempotent values, which should perhaps be added this way to prevent them from being in datum if idempotent middleware is omitted).